python/ambassador/ir/irutils.py

```diff
@@ -181,14 +184,21 @@ def selector_matches(
         logger.debug("    no incoming labels => False")
         return False
 
-    selmatch = False
+    # Ambassador (2.0-2.3) & (3.0-3.1) consider a match on a single label as a "good enough" match.
+    # In versions 2.4+ and 3.2+ _ALL_ labels in a selector must be present for it to be considered a match.
+    # DISABLE_STRICT_LABEL_SELECTORS provides a way to restore the old unintended loose matching behaviour
+    # in the event that it is desired. The ability to disable strict label matching will be removed in a future version.
+    disable_strict_selectors = parse_bool(os.environ.get("DISABLE_STRICT_LABEL_SELECTORS", "false"))
```

Comment:
Part of me thinks the "correct" fix is to move this code to Go so we can just call the standard Kubernetes library function for checking label selectors.